packages/framework/agent-scheduler/src/scheduler.ts

```diff
@@ -420,8 +422,11 @@ export class AgentSchedulerFactory implements IFluidDataStoreFactory {
 
     public static async createChildInstance(parentContext: IFluidDataStoreContext): Promise<AgentScheduler> {
         const packagePath = [...parentContext.packagePath, AgentSchedulerFactory.type];
-        const router = await parentContext.containerRuntime.createDataStore(packagePath);
-        return requestFluidObject<AgentScheduler>(router, "/");
+        const dataStore = await parentContext.containerRuntime.createDataStore(packagePath);
+        const maybeHandle: FluidObject<IFluidHandle> = (dataStore as any);
```

Comment:
why do we need to make this cast? This suggests we are doing something wrong - ideally object we return should have right signature shape not to need to do these casts

Reply:
So, two things:

- As Tony pointed out in another comment, there was a cleaner way of doing this cast and keep it working: `const maybeHandle = dataStore as FluidObject<IFluidHandle>;`.
- It's still necesary to cast if we want to go with the approach of (for now) having the `IFluidHandle` be present in the concrete classes but not on the interfaces that are usually in the API signatures, so as not to introduce any surface API changes yet, but start to plumb the usage of the new pattern internally (we know that the underlying objects might already expose a handle, so we start trying to use it if it's there) and make sure everything works.

When we expose the handle explicitly as part of the API, these casts should definitely go away. I can add TODO comments to point out the need to remove them once it's appropriate.

Reply:
our interfaces are confused between IFluidHandle and IFluidLoadable, i'm actually thinking about what the right one to use here is.

Reply:
As long as it's temporary, I'm fine either way. But let's have a plan not to require any casting in user code